utils.py

```diff
@@ -30,6 +30,36 @@
 logger.setLevel(logging.DEBUG)
 
 
+class DenseTensor(object):
+    def __eq__(self, other):
+        if isinstance(other, DenseTensor):
+            return True
+        else:
+            return False
+
+
+class SparseTensor(object):
+    """ Sparse tensor data format.
+    Parameters
+    ----------
+    mode_formats: a list containing the format of each mode ("dense" or "compressed")
```

Comment:
Any drawback on using a similar notion as pytaco?
They have pt.dense/pt.compressed, can we have ad.format.dense/ad.format.sparse as types? I would discourage the usage of strings.

Reply:
Yes we can, but I just felt string can do the same thing. What's the reason that you think using string is bad?

Reply:
It's easy to put an invalid string in the args, namely, it need run time check (string comparison). So if the arg is a type, it's easy for the user to select a list of types, for example, you can autocomplete a tensor type when writing the code. 

(As I realize python needs run-time check anyway, so there is no static analysis for simple type checking like C++. Though the still, API design should follow the principle that it's hard for the user to use it wrong (provide invalid input).

Reply:
OK, sounds good

Reply:
changes made, please take a look